src/OrchardCore/OrchardCore.ContentLocalization.Abstractions/Records/Migrations.cs

```diff
@@ -45,5 +52,19 @@ public int UpdateFrom1()
 
             return 2;
         }
+
+        // This code can be removed in a later version.
+        public int UpdateFrom2()
+        {
+            SchemaBuilder.AlterTable(nameof(LocalizedContentItemIndex), table => table
+                .AddColumn<bool>("Latest", c => c.WithDefault(false))
+            );
+
+            SchemaBuilder.AlterTable(nameof(LocalizedContentItemIndex), table => table
+                .CreateIndex("IDX_LocalizationPartIndex_State", "Published", "Latest")
```

Comment:
Because the query is against the index the internal index needs all properties to be used
so it needs an index for
`Published`, `Latest`, `ContentItemId` 

and another index for
`Published`, Latest`, `LocalizationSet`

Otherwise SQL Server will not use the index, but fall back to a clustered scan (so it is useless). many of our indexes have this issue... slowly trying to fix them.

For infos

If it was a `Query<ContentItem, LocalizationPartIndex>(x => predicate)` it should also include the `DocumentId` as the first property on the index, as that gets used for the join part of the sql predicate.

They also have to be in order of how they are being queried.

Reply:
Okay, just did the same as for autoroute migrations when the new `Latest` column was added

            SchemaBuilder.AlterIndexTable<AutoroutePartIndex>(table => table
                .CreateIndex("IDX_AutoroutePartIndex_State", "Published", "Latest")
            );

Hmm, but there were already these indexes


            SchemaBuilder.AlterIndexTable<LocalizedContentItemIndex>(table => table
                .CreateIndex("IDX_LocalizationPartIndex_LocalizationSet_Culture", new[] { "LocalizationSet", "Culture" })
            );

            SchemaBuilder.AlterIndexTable<LocalizedContentItemIndex>(table => table
                .CreateIndex("IDX_LocalizationPartIndex_ContentItemId", "ContentItemId")
            );

            // In a migration step
            SchemaBuilder.AlterIndexTable<LocalizedContentItemIndex>(table => table
                .AddColumn<bool>(nameof(LocalizedContentItemIndex.Published)));

So, do we need to drop some indexes before adding new ones ?

**Maybe simpler that you update the PR with what you are suggesting**

Reply:
No problem, I'll update it.

Yes the existing indexes are not the best, I've been avoiding Dropping indexes, because it's been tricky with the one time I did that on Autoroute (doesn't work on SQL Server 2014 and below). So have been mostly, just improving them on new setups, and trying to get the new ones we're adding to be good.

Reply:
Okay cool then, thanks

Just need to fix unit tests before leaving

Reply:
Okay it was just a build error on the new `AlterIndexTable`

Feel free to update the PR